tool/internal/setup/setup.go

```diff
@@ -78,26 +81,30 @@ func Setup(logger *slog.Logger) error {
 }
 
 // BuildWithToolexec builds the project with the toolexec mode
-func BuildWithToolexec(logger *slog.Logger, args []string) error {
+func BuildWithToolexec(ctx context.Context, args []string) error {
+	logger := util.LoggerFromContext(ctx)
+
 	// Add -toolexec=otel to the original build command and run it
 	execPath, err := os.Executable()
 	if err != nil {
 		return ex.Errorf(err, "failed to get executable path")
 	}
-	insert := "-toolexec=" + execPath
-	newArgs := make([]string, 0, len(args)+1) // Avoid in-place modification
-	newArgs = append(newArgs, args[:2]...)    // Add "go build"
-	newArgs = append(newArgs, insert)         // Add "-toolexec=..."
-	newArgs = append(newArgs, args[2:]...)    // Add the rest
-	logger.Info("Running go build with toolexec", "args", newArgs)
+	insert := "-toolexec=" + execPath + " --work-dir=" + util.GetOtelWorkDir() + " toolexec"
```

Comment:
One final thing, this fix is unnecessary because we pass it via environment later 
```
util.Assert(pwd != "", "invalid working directory")
	env = append(env, fmt.Sprintf("%s=%s", util.EnvOtelWorkDir, pwd))
```
Just get it from env var.